apps/shipments/permissions.py

```diff
@@ -1,40 +1,43 @@
 from datetime import datetime, timezone
+
 from django.core.exceptions import ObjectDoesNotExist
 from rest_framework import permissions
-from .models import PermissionLink, Shipment
-
-
-class IsOwner(permissions.BasePermission):
-    """
-    Custom permission to only allow owners of an object to edit it
-    """
 
-    def has_object_permission(self, request, view, obj):
-
-        # Permissions are only allowed to the owner of the shipment.
-        return obj.owner_id == request.user.id
+from apps.permissions import is_owner
+from .models import PermissionLink, Shipment
 
 
 class IsShipmentOwner(permissions.BasePermission):
     """
-    Custom permission to only allow owners of an object to edit it
+    Custom permission to only allow owners of an object to edit it (for use in related querysets)
     """
-
     def has_permission(self, request, view):
         shipment = Shipment.objects.get(pk=view.kwargs['shipment_pk'])
+        return is_owner(request, shipment)
 
-        # Permissions are only allowed to the owner of the shipment.
-        return shipment.owner_id == request.user.id
 
+class IsListenerOwner(permissions.BasePermission):
```

Comment:
Considering this is in both Eth and Jobs, wouldn't it make sense to move this to the base apps.permission folder?

Reply:
I debated doing that, but I think it makes sense to keep this in Shipments permissions since it tests whether a listener has access to a Shipment